### PR TITLE
HDF5: Add develop Version

### PR DESCRIPTION
**Pull request by ax3l**
Add support for building the development branch of HDF5.

### CODE REVIEW

File: var/spack/repos/builtin/packages/hdf5/package.py [hunk 1]
```diff
@@ -106,6 +114,11 @@ def url_for_version(self, version):
         url = "https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-{0}/hdf5-{1}/src/hdf5-{1}.tar.gz"
         return url.format(version.up_to(2), version)
 
+    @when('@develop')
+    def autoreconf(self, spec, prefix):
+        sh = which('sh')
+        sh('autogen.sh')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/hdf5/package.py]: Does `Executable('./autogen.sh')` not suffice?

Comment (ax3l) [reply var/spack/repos/builtin/packages/hdf5/package.py]: That's a thing?! :)

Comment (ax3l) [reply var/spack/repos/builtin/packages/hdf5/package.py]: No, doesn't work:
```
==> Executing phase: 'autoreconf'
==> Error: RuntimeError: configure script not found in /home/axel/src/spack/var/spack/stage/hdf5-develop-f5u36aysfttnkeww2jmqzrgsl5qa3xpl/hdf5

/home/axel/src/spack/lib/spack/spack/build_systems/autotools.py:220, in set_configure_or_die:
        217        """
        218        # Check if a configure script is there. If not raise a RuntimeError.
        219        if not os.path.exists(self.configure_abs_path):
  >>    220            msg = 'configure script not found in {0}'
        221            raise RuntimeError(msg.format(self.configure_directory))
        222
        223        # Monkey-patch the configure script in the corresponding module
```

Don't understand why, it does have `+x` when I check...

Comment (tgamblin) [reply var/spack/repos/builtin/packages/hdf5/package.py]: Well, you need:

```python
autogen = Executable('./autogen.sh')
autogen()
```

Did you call it?

Comment (ax3l) [reply var/spack/repos/builtin/packages/hdf5/package.py]: ha :D

Comment (ax3l) [reply var/spack/repos/builtin/packages/hdf5/package.py]: @tgamblin updated :) thanks for the hint!

**Pull request merged**